### PR TITLE
After restarting application, the Others tab shows that Auth is automatically signed out

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the UWP implementation?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

**Repro Steps**

1. Open the ACFDemo test app and go to Others tab
2. Click "Test MBaaS" button and Sign in your social account
3. Check if the Auth has successfully signed in and it shows "User  authenticated"
4. Restart the test app and go to Others tab
5. Check if the Auth is automatically signed out and it shows "User not authenticated"

**Expected Behaviour**
After restarting application, the Auth should still signed in on the Others tab
 
**Actual Behaviour**
After restarting application, the Others tab shows that Auth is automatically signed out

## Related PRs or issues

[AB#63530](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/63530)

### CODE REVIEW

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ protected override async void OnAppearing()
             DistributeEnabledSwitchCell.IsEnabled = acEnabled;
             PushEnabledSwitchCell.On = await Push.IsEnabledAsync();
             PushEnabledSwitchCell.IsEnabled = acEnabled;
-            if (userInfo?.AccountId != null)
+            if (Application.Current.Properties.ContainsKey("accountId") && Application.Current.Properties["accountId"] is string id)
```
Comment (MatkovIvan) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]: Move `"accountId"` to const

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ namespace Contoso.Forms.Demo
 {
     using XamarinDevice = Xamarin.Forms.Device;
 
+    private const string ACCOUNT_ID = "accountId";
```
Comment (MatkovIvan) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]: Please move const to the class and rename it to PascalCase

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 3]
```diff
@@ -126,6 +129,7 @@ async void SignInInformation(object sender, EventArgs e)
         {
             if (userInfo != null)
             {
+                Application.Current.Properties[AccountId] = null;
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]: Shouldn't this be in signOut instead? I don't understand why we would lose context when looking at sign in details.

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 4]
```diff
@@ -17,7 +17,8 @@ namespace Contoso.Forms.Demo
 
     [Android.Runtime.Preserve(AllMembers = true)]
     public partial class OthersContentPage
-    { 
+    {
+
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]:
```suggestion
```

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 5]
```diff
@@ -27,7 +28,8 @@ static OthersContentPage()
         }
 
         private UserInformation userInfo;
-	
+        private const string AccountId = "accountId";
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]:
```suggestion

        private const string AccountId = "accountId";
```

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 6]
```diff
@@ -47,11 +49,18 @@ protected override async void OnAppearing()
...
+                SignInInformationButton.Text = "Authentication status unknown";
+            }
+            else if (Application.Current.Properties[AccountId] is string)
             {
                 SignInInformationButton.Text = "User authenticated";
             }
-            else SignInInformationButton.Text = "User not authenticated";
+            else
+            {
+                SignInInformationButton.Text = "User not authenticated";
```
Comment (jaelim-ms) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]:
```suggestion
                SignInInformationButton.Text = "User is not authenticated";
```

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs [hunk 7]
```diff
@@ -47,11 +49,18 @@ protected override async void OnAppearing()
...
             AuthEnabledSwitchCell.On = await Auth.IsEnabledAsync();
             AuthEnabledSwitchCell.IsEnabled = acEnabled;
-            if (userInfo?.AccountId != null)
+            if (!Application.Current.Properties.ContainsKey(AccountId))
+            {
+                SignInInformationButton.Text = "Authentication status unknown";
+            }
+            else if (Application.Current.Properties[AccountId] is string)
             {
                 SignInInformationButton.Text = "User authenticated";
```
Comment (jaelim-ms) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/ModulePages/OthersContentPage.xaml.cs]:
```suggestion
                SignInInformationButton.Text = "User is authenticated";
```

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs [hunk 8]
```diff
@@ -57,11 +59,18 @@ protected override async void OnAppearing()
...
             EventFilterEnabledSwitchCell.On = _eventFilterStarted && await EventFilterHolder.Implementation?.IsEnabledAsync();
             EventFilterEnabledSwitchCell.IsEnabled = acEnabled && EventFilterHolder.Implementation != null;
-            if (userInfo?.AccountId != null)
+            if (!Application.Current.Properties.ContainsKey(AccountId))
+            {
+                SignInInformationButton.Text = "Authentication status unknown";
+            }
+            else if (Application.Current.Properties[AccountId] is string)
             {
                 SignInInformationButton.Text = "User authenticated";
```
Comment (jaelim-ms) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs]:
```suggestion
                SignInInformationButton.Text = "User is authenticated";
```

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs [hunk 9]
```diff
@@ -57,11 +59,18 @@ protected override async void OnAppearing()
...
             }
-            else SignInInformationButton.Text = "User not authenticated";
+            else
+            {
+                SignInInformationButton.Text = "User not authenticated";
```
Comment (jaelim-ms) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs]:
```suggestion
                SignInInformationButton.Text = "User is not authenticated";
```

**Pull request merged**